relay-general/src/protocol/contexts/device.rs

```diff
@@ -1,8 +1,27 @@
+use std::collections::HashMap;
+
+use once_cell::sync::Lazy;
+
 use crate::protocol::FromUserAgentInfo;
 use crate::store::user_agent::is_known;
 use crate::types::{Annotated, Object, Value};
 use crate::user_agent::{parse_device, ClientHints};
 
+/// A hashmap that translates from the android model to the more human-friendly product-names.
+/// E.g. NE2211 -> OnePlus 10 Pro 5G
+pub static ANDROID_MAP: Lazy<HashMap<&'static str, &'static str>> = Lazy::new(|| {
+    let mut map = HashMap::new();
+    let android_str = include_str!("../../../../files/android_models.csv");
+
+    for line in android_str.lines() {
+        let fields: Vec<&str> = line.split(',').collect();
+        if fields.len() >= 4 {
+            map.insert(fields[3].trim(), fields[1].trim());
```

Comment:
Currently, the first line (the header) is also included in the map I believe, and reordering of columns can break this. I think using the first line and extracting the column indices from there is more solid -- I'd look at the indices of `Device` and `Model`, so if Google inserts another column at the beginning, this will still work. I believe changing the column name is less likely than introducing new column names or reordering them.